tests/dts/unit/cases/plugins.ts

```diff
@@ -16,7 +16,7 @@ const plugin: prettier.Plugin<PluginAST> = {
   ],
   parsers: {
     lines: {
-      parse(text, options) {
+      parse(text, options = {}) {
```

Comment:
Why?

Reply:
If the second parameter doesn't have a default argument, TypeScript thinks it's mandatory, and thus thinks that `parse` here doesn't provide the first overload in the definition of `Parser` - namely:

```
parse(text: string): T | Promise<T>;
```

You can see this locally by deleting the ` = {}` that I add here and running:

```
yarn build && NODE_ENV=production yarn test dts
```

- it'll fail.

Reply:
options alway provide when calling inside prettier. we allow it omitted only because we crushed jest once. not really meaningful.

Reply:
If you want to enforce that `options` is mandatory, we can do that. I figured that it shouldn't be mandatory because the test that I'm restoring here was specifically testing that we could call `parse(text)` without a second argument, and so I assumed that was something important to support and that the types were wrong. If actually it's the _test_ that's wrong, I can leave the types alone and change the test instead to pass an `options` argument.

Reply:
I don't really know TS well, but if it's for bultin parsers, then yes, we should force user provide options, since we change parsers  anytime we want, user call it manually is not encouraged, they are on their own risk, though we allow omit the `options` but we don't want document/add type for this.

Reply:
👍 - I'll rework this PR tomorrow morning to modify the test instead of the types.

Reply:
In short, built-in plugins are considered private, the parse function should only be called by prettier, and we always provide `options`.